apps/back/src/Controller/UserController.php

```diff
@@ -4,52 +4,100 @@
 
 namespace App\Controller;
 
+use App\Dto\UserDto;
 use App\Entity\User;
 use App\Repository\UserRepository;
 use Doctrine\ORM\EntityManagerInterface;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
-use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 use Symfony\Component\Routing\Annotation\Route;
-use Symfony\Component\Security\Http\Attribute\CurrentUser;
 
-class UserController
+final class UserController
 {
     public function __construct(
         private readonly EntityManagerInterface $entityManager,
         private readonly UserRepository $userRepository,
+        private readonly MailerInterface $mailer,
     ) {
     }
 
-    /**  This route demonstrate the usage of role_right */
-    #[Route('/user', name: 'user_put', methods: ['PUT'])]
-    #[IsGranted('ROLE_RIGHT_USER_CREATE')]
-    public function login(Request $request): JsonResponse
+    #[Route('/user', name: 'create_user', methods: ['POST'])]
+    public function createUser(Request $request, UserPasswordHasherInterface $passwordHasher): JsonResponse
     {
-        $data = \json_decode((string) $request->getContent(), true, 512);
-        if (!$data) {
-            throw new BadRequestException('data is not valid');
+        $userDto = new UserDto(json_decode($request->getContent(), true));
+
+        if ($this->userRepository->findOneBy(['email' => $userDto->getEmail()])) {
+            return new JsonResponse('Email already exists', 400);
         }
 
-        if (!\is_string($data['username'])) {
-            throw new BadRequestException('username must be a string');
+        $user = new User($userDto->getEmail());
+        $user->setPassword($passwordHasher->hashPassword($user, $userDto->getPassword()));
+        $this->entityManager->persist($user);
+        $this->entityManager->flush();
+
+        $email = (new TemplatedEmail())
```

Comment:
do it in a separate object.
A good option would be

$mailer->sendRegistrationMail($userEntity);